app/src/main/java/org/thoughtcrime/securesms/util/IP2Country.kt

```diff
@@ -5,59 +5,47 @@ import android.content.Context
 import android.content.Intent
 import android.content.IntentFilter
 import androidx.localbroadcastmanager.content.LocalBroadcastManager
-import org.session.libsignal.utilities.Log
 import com.opencsv.CSVReader
+import dagger.hilt.android.qualifiers.ApplicationContext
 import org.session.libsession.snode.OnionRequestAPI
+import org.session.libsession.utilities.binarySearchLastAndGet
+import org.session.libsignal.utilities.Log
 import org.session.libsignal.utilities.ThreadUtils
 import java.io.File
 import java.io.FileOutputStream
 import java.io.FileReader
+import javax.inject.Inject
+import javax.inject.Singleton
 
-class IP2Country private constructor(private val context: Context) {
+@Singleton
+class IP2Country @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
     private val pathsBuiltEventReceiver: BroadcastReceiver
-    val countryNamesCache = mutableMapOf<String, String>()
+    val countryNamesCache = mutableMapOf<String, String?>()
 
-    private fun Ipv4Int(ip:String) = ip.takeWhile { it != '/' }.split('.').foldIndexed(0L) { i, acc, s ->
-        val asInt = s.toLong()
-        acc + (asInt shl (8 * (3-i)))
-    }
+    private fun ipv4Int(ip: String) =
+        ip.takeWhile { it != '/' }.split('.').reversed()
+            .fold(0L) { acc, octet -> acc shl 8 or octet.toLong() }
```

Comment:
Those two functions are not equivalent.
You can test out their outputs and they will output different things.
Was the original broken? If not the new function is not the same.